src/util/context.h

```diff
@@ -38,6 +38,11 @@ class contextt
   explicit contextt(const messaget &msg) : msg(msg)
   {
   }
+  contextt& operator=(const contextt& t)
+  {
+    // copy assignment operator for old frontend typechecking
+    return *this;
```

Comment:
Having a no-op copy-assignment operator is a Bad Idea. Why is it a no-op?

Reply:
Indeed, an no-op copy-assignment opterator is very bad. 

This copy-assignment operator was deliberately made no-op to satisfy the copy operation in `ansi_c_typecheck` function in `src/ansi-c/ansi_c_typecheck.cpp`: 
```
 37 contextt context1(message_handler), context2(message_handler);
 38   ansi_c_parse_treet ansi_c_parse_tree;
 39
 40   context1 = ns.get_context();
```
The copy assignment operation used to work but got broken by the const reference `const messaget &msg;` introduced when centralising the message handler in commit 4dc847814c2. Essentially, this copy-assignment operator restores the semantics of the operation in line 40. (And in my humble opinion, the original usage of default copy-assignment operator was not a good idea either... but it works). 

If we wanted a proper copy-assignment operator of context class for the old frontends, it would be opening a can of worms. Because we would also need to deal with the associated data structures such as symbolst, ordered_symbolst ... etc. 

Apologies for the hacky way used in this patch, but I reckon this is perhaps the most efficient way to restore the old C frontend in a timely manner, as well as avoid opening a can of worms.

Reply:
I see, many thanks for the explanations. Your answer would make an excellent comment for the operator=. ;)
Do you think it is easier not having `contextt` copyable?

Reply:
>Do you think it is easier not having contextt copyable?
Yes, I believe it's currently doing so. When using new C frontend, `contextt` class has a const ref member, but does not have a copy assignment operator or copy constructor. The compiler will report an error like "Use of deleted copy assignment operator" (or something similar) when someone writes a line of code like line 40. 

Additionally, "=delete" has been added to make it more explicit when reading the code, as well as for confomance to the rule of three.

Reply:
So, what about a comment of explanation on the no-op operator=? E.g. your reply above. No one is going to check github in order to understand why it is a no-op.

Reply:
Done. 
Added a comment to explain the hacky code.